### PR TITLE
introduce different travis runners for different jobs

### PR DESCRIPTION
**Pull request by lmm-git**

### CODE REVIEW

**Comment by lmm-git** [comment]
Please review; should I squash the commits?

**Comment by lmm-git** [comment]
probably @TimNN

File: notes/deploy.sh [hunk 1]
```diff
@@ -73,4 +63,16 @@ git config user.name "Travis CI"
 git config user.email "$AUTHOR_EMAIL"
 git diff-index --quiet HEAD || git commit -m 'Updating Documentation'
 git checkout -b "$DEPLOY_BRANCH"
+
+# skip deploy if we are not on our master branch
+if [ "$TRAVIS_BRANCH" != "master" ]; then
+    msg "not on master, skipping documentation build"
```
Comment (TimNN) [inline notes/deploy.sh]: Please adjust the log message that only deployment is skipped

File: notes/deploy.sh [hunk 2]
```diff
@@ -73,4 +63,16 @@ git config user.name "Travis CI"
...
+    exit 0
+fi
+
+if [ "$TRAVIS_PULL_REQUEST" != "false" ]; then
+    msg "building a pull request, skipping documentation build"
```
Comment (TimNN) [inline notes/deploy.sh]: Same here

File: notes/deploy.sh [hunk 3]
```diff
@@ -73,4 +63,16 @@ git config user.name "Travis CI"
 git config user.email "$AUTHOR_EMAIL"
 git diff-index --quiet HEAD || git commit -m 'Updating Documentation'
 git checkout -b "$DEPLOY_BRANCH"
+
+# skip deploy if we are not on our master branch
```
Comment (TimNN) [inline notes/deploy.sh]: I believe we could (and should) move this above all the git commands (before the `log "going to deploy to"`)

**Comment by TimNN** [comment]
Also, could you add a few blank lines in the `matrix` section in `travis.yml`

**Comment by TimNN** [comment]
Looks good so far. Could you please make sure that `log "going to deploy to $BOLD$REPO_URL$RESET"` is not executed if deployment would be skipped?

Also, feel free to comment once you made any changes because GitHub does not send emails when new commits are added to a PR.

**Comment by lmm-git** [comment]
should be fixed now? Shall I squash?

**Comment by TimNN** [comment]
I think GitHub can do the squashing, let me try

**Pull request merged**